### PR TITLE
upgrade vue-resource to 1.0.1

### PR DESCRIPTION
**Pull request by deiucanta**
This solves the `response.data` problem but not the profile photo upload.

### CODE REVIEW

**Comment by taylorotwell** [comment]
OK, we can't release until the profile photo upload is fixed so we'll have to look into that.

File: resources/assets/js/interceptors.js [hunk 1]
```diff
@@ -1,22 +1,8 @@
-module.exports = {
-    /**
-     * Intercept the outgoing requests.
-     *
-     * Set common headers on the request.
-     */
-    request(request) {
-        request.headers['X-XSRF-TOKEN'] = Cookies.get('XSRF-TOKEN');
+module.exports = (request, next) => {
+    console.log(request.headers);
```
Comment (RSchwan) [inline resources/assets/js/interceptors.js:2]: Is this intended?

Comment (themsaid) [reply resources/assets/js/interceptors.js:2]: Please remove this.

File: resources/assets/js/interceptors.js [hunk 2]
```diff
@@ -1,22 +1,8 @@
...
 
-        return request;
-    },
-
-
-    /**
-     * Intercept the incoming responses.
-     *
-     * Handle any unexpected HTTP errors and pop up modals, etc.
-     */
```
Comment (themsaid) [inline resources/assets/js/interceptors.js:18]: Please keep this comment.

File: resources/assets/js/interceptors.js [hunk 3]
```diff
@@ -28,6 +14,14 @@ module.exports = {
...
+        if (response.headers.get('content-type') === 'application/json') {
+            response.json().then(data => {
+                response.data = data;
+            });
+        } else {
+            response.text().then(data => {
+                response.data = data;
+            });
+        }
+    });
```
Comment (themsaid) [inline resources/assets/js/interceptors.js:26]: What's the purpose of this code? `data` is already available in the response object.

**Pull request closed without merging**